### PR TITLE
Add slow-speed timeout when downloading mod

### PR DESCRIPTION
**Pull request by Alystrasz**
Adds a timeout if connection is too slow.

***

Previous PR text:

&nbsp;

While currently in a business trip, not having access to a super duper fiber Internet connection, I tried downloading some mods, which failed due to the poor Internet connection: mod request would timeout before mod is downloaded.

```text
[2023-12-02] [11:37:16] [NORTHSTAR] [info] Mods list successfully fetched.
[2023-12-02] [11:37:16] [NORTHSTAR] [info] Loading mods configuration...
[2023-12-02] [11:37:16] [NORTHSTAR] [info] ==> Loaded configuration for mod "Odd.s2space"
[2023-12-02] [11:37:16] [NORTHSTAR] [info] Done loading verified mods list.
[2023-12-02] [11:37:32] [NORTHSTAR] [info] Fetching mod archive from https://gcdn.thunderstore.io/live/repository/packages/odds-s2space-0.0.5.zip
[2023-12-02] [11:37:32] [NORTHSTAR] [info] Downloading archive to C:/users/steamuser/Temp/odds-s2space-0.0.5.zip
[2023-12-02] [11:38:02] [NORTHSTAR] [error] Fetching mod archive failed: Timeout was reached
[2023-12-02] [11:38:02] [NORTHSTAR] [error] Something went wrong while fetching archive, aborting.
```

Looking in the code, I realized I put the same timeout for the mod fetching request than for the mods list fetching request, which is **30 seconds**.

I thought about setting the timeout with a higher value (a few minutes or so), but then I remembered when I was a little boy, I would leave my computer running overnight to download games because my Internet connection was so shitty :>

In this PR, I ended up completely removing the timeout attribute, meaning mod fetching requests will never time out (according to [curl documentation](https://curl.se/libcurl/c/CURLOPT_TIMEOUT.html)).

```text
[2023-12-02] [12:50:45] [NORTHSTAR] [info] Mods list successfully fetched.
[2023-12-02] [12:50:45] [NORTHSTAR] [info] Loading mods configuration...
[2023-12-02] [12:50:45] [NORTHSTAR] [info] ==> Loaded configuration for mod "Odd.s2space"
[2023-12-02] [12:50:45] [NORTHSTAR] [info] Done loading verified mods list.
[2023-12-02] [12:50:56] [NORTHSTAR] [info] Fetching mod archive from https://gcdn.thunderstore.io/live/repository/packages/odds-s2space-0.0.5.zip
[2023-12-02] [12:50:56] [NORTHSTAR] [info] Downloading archive to C:/users/steamuser/Temp/odds-s2space-0.0.5.zip
[2023-12-02] [12:51:48] [NORTHSTAR] [info] Mod archive successfully fetched.
```

/storytelling off

### CODE REVIEW

**Comment by Alystrasz** [comment]
This is [not straightforward](https://curl.se/docs/faq.html#Why_does_not_curl_return_an_erro) to implement, but I suppose I could implement such a behavior using [CURLOPT_LOW_SPEED_TIME](https://curl.se/libcurl/c/CURLOPT_LOW_SPEED_TIME.html).
Would you like me to do that in this PR?

File: NorthstarDLL/mods/autodownload/moddownloader.cpp [hunk 1]
```diff
@@ -132,9 +132,13 @@ void FetchModSync(std::promise<std::optional<fs::path>>&& p, std::string_view ur
 	CURL* easyhandle;
 	easyhandle = curl_easy_init();
 
-	curl_easy_setopt(easyhandle, CURLOPT_TIMEOUT, 30L);
 	curl_easy_setopt(easyhandle, CURLOPT_URL, url.data());
 	curl_easy_setopt(easyhandle, CURLOPT_FAILONERROR, 1L);
+
+	/* abort if slower than 30 bytes/sec during 10 seconds */
```
Comment (GeckoEidechse) [inline NorthstarDLL/mods/autodownload/moddownloader.cpp:138]: We only use `//` for single line comments AFAIK

```suggestion
	// abort if slower than 30 bytes/sec during 10 seconds
```

**Comment by GeckoEidechse** [comment]
For testing I guess the easiest way would be to yank the ethernet cable half way through download?

Not sure if we even need testing for it though as the feature as a whole is still considered unreleased as it's gated behind convar.

**Comment by Alystrasz** [comment]
> For testing I guess the easiest way would be to yank the ethernet cable half way through download?

I did the testing by cutting the Wi-Fi signal, but it's the same idea, yeah.

**Comment by GeckoEidechse** [comment]
Given that you already tested it, I'm skip testing on my end and just merge directly instead ^^

**Comment by GeckoEidechse** [comment]
wait, where did the arg removal go, hol' up.

**Comment by GeckoEidechse** [comment]
Oh wait, so the change from c656859bcf8164e6c078e61fa9a95ca9b145004c was already merged in #595. Guess I'm renaming this PR here then so it's a bit better reflecting of the actual change, lol

**Comment by GeckoEidechse** [comment]
Okay, now it should be ready to merge, lol

**Pull request merged**